File: 백준/Silver/13305. 주유소/주유소.py
N = int(input()) # 도시의 개수
dis = list(map(int, input().split())) # 각 도시까지의 거리
oil = list(map(int, input().split())) # 각 주요소의 가격
min_oil = oil[0]

ans = dis[0] * oil[0]

for cur in range(1, N-1):
    if min_oil > oil[cur]:
        min_oil = oil[cur]
    ans += min_oil * dis[cur]

print(ans)

File: 백준/Silver/13305. 주유소/주유소.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int N = sc.nextInt();
        
        int[] dis = new int[N - 1];
        for (int i = 0; i < N - 1; i++) {
            dis[i] = sc.nextInt();
        }
        
        int[] oil = new int[N];
        for (int i = 0; i < N; i++) {
            oil[i] = sc.nextInt();
        }
        
        int minOil = oil[0]; 
        long ans = (long) dis[0] * minOil; 
        
        for (int cur = 1; cur < N - 1; cur++) {
            if (minOil > oil[cur]) {
                minOil = oil[cur]; 
            }
            ans += (long) minOil * dis[cur]; 
        }
        
        System.out.println(ans);

        sc.close();
    }
}